src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(last, value, null);
+        last.next = newNode;
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            addFirst(value);
+            return;
+        }
+        checkIndex(index - 1);
+        Node<T> current = getNodeByIndex(index);
+        Node<T> prevNode = current.prev;
+        Node<T> newNode = new Node<>(prevNode, value, current);
+        prevNode.next = newNode;
+        current.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = getNodeByIndex(index);
+        T deletedValue = current.item;
+        current.item = value;
+        return deletedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = getNodeByIndex(index);
+        if (size == 1) {
+            removeLast();
+            return current.item;
+        }
+        return unlink(current);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> current = first;
+        while (current != null
+                && !(current.item == object
+                   || current.item != null && current.item.equals(object))) {
```

Comment:
can we write this while loop so we could look through all nodes and if we find our node by an object we could call unlink inside of this while loop and after that return true plus instead of writing this complex condition at the end of the method we could write return false